### PR TITLE
apply serialize pattern to the code

### PR DESCRIPTION
**Pull request by bedreski**
Code refactored with Serializer design pattern, where it is useful

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -4,58 +4,22 @@ class TasksController < ApplicationController
...
-        'userId' => user.id,
-        'status' => task.status,
-        'name' => task.name,
-        'createdAt' => task.created_at,
-        'completedAt' => task.completed_at
-      }
-    end
-
-    render(json: serialized_tasks)
+    render(json: tasks.map { |task| TaskSerializer.new(task).as_json })
```
Comment (EduardoSCosta) [inline app/controllers/tasks_controller.rb:7]: você pode atribuir as tasks a uma variavel:
```suggestion
    serialized_tasks = tasks.map { |task| TaskSerializer.new(task).as_json }
    
    render(json: serialized_tasks)
```

File: spec/requests/tasks/complete_spec.rb [hunk 2]
```diff
@@ -76,7 +76,7 @@
 
       context 'when user has notification enabled to email, SMS, Telegram and Whatsapp' do
         let(:user) { create(:user, notification_preferences: { task_completed: %i[email sms telegram whatsapp] }) }
-        let(:task) { create(:task, user:) }
+        let(:task) { create(:task) }
```
Comment (EduardoSCosta) [inline spec/requests/tasks/complete_spec.rb:79]:
```suggestion
        let(:task) { create(:task, user: user) }
```

Comment (bedreski) [reply spec/requests/tasks/complete_spec.rb:79]: Yes, sir

File: app/controllers/tasks_controller.rb (hunk 1 above)
Comment (bedreski) [reply app/controllers/tasks_controller.rb]: Yes, sir

**Review by danielmbrasil — APPROVED** [review]

**Pull request merged**